signer/aws/aws.go

```diff
@@ -25,51 +27,74 @@ import (
 // OpenSearchService AWS OpenSearchService Name
 const OpenSearchService = "es"
 
-// Signer is a interface that will implement opensearchtransport.Signer
+// OpenSearchServiceServerless AWS OpenSearchService Serverless Name
+const OpenSearchServiceServerless = "aoss"
+
+const emptyBodySHA256 = "2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824"
+
+// Signer is an interface that will implement opensearchtransport.Signer
 type Signer struct {
 	session session.Session
 	service string
 }
 
-// NewSigner returns an instance of Signer for AWS OpenSearchService
+// NewSigner returns an instance of Signer for configured for AWS OpenSearchService.
```

Comment:
The correct names would be "Amazon OpenSearch Service" and "Amazon OpenSearch Serverless", maybe we can update?

Reply:
Will address